compose/config/types.py

```diff
@@ -163,3 +163,9 @@ def parse(cls, volume_config):
     def repr(self):
         external = self.external + ':' if self.external else ''
         return '{ext}{v.internal}:{v.mode}'.format(ext=external, v=self)
+
+    @property
+    def is_named_volume(self):
+        return self.external and not (
+            self.external.startswith('.') or self.external.startswith('/')
```

Comment:
Thanks!
However, `split_path_mapping` tests for a relative path, so it's slightly different.
Also, the paths in `VolumeSpec` are already normalized for the docker engine, so we don't need to consider OS variance.